src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx

```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
+    });
+    waitFor(() => {
+      const installedTabContent = screen.getByText('Installed');
+      expect(installedTabContent).toBeInTheDocument();
+    });
+    waitFor(() => {
+      const availableTabContent = screen.queryByText('Available Plugin');
+      expect(availableTabContent).toBeNull();
+    });
+  });
+
+  test('installs a plugin when "Install" button is clicked', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Install'));
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Install')).toBeInTheDocument();
+    });
+  });
+
+  test('modifies a plugin when "Modify" button is clicked', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Modify'));
+    });
+    waitFor(() => {
+      expect(screen.getByText('Modify')).toBeInTheDocument();
+    });
+  });
+
+  test('displays the organization screen with the correct title', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      expect(screen.getByTestId('organization-screen')).toBeInTheDocument();
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Plugin Store')).toBeInTheDocument();
+    });
+  });
+
+  test('handles API error and displays an error message', async () => {
+    jest.mock('@apollo/client', () => ({
+      ...jest.requireActual('@apollo/client'),
+      useQuery: jest.fn(() => ({
+        data: undefined,
+        loading: false,
+        error: new Error('API error'),
+      })),
+    }));
+
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
```

Comment:
I don't think we need to use `waitFor` here?

Reply:
I am using waitFor here to make sure the data gets fetched as per the code,